frontend/public/components/monitoring/dashboards/index.tsx

```diff
@@ -370,98 +360,94 @@ const Card: React.FC<CardProps> = ({ panel }) => {
+
   React.useEffect(() => {
-    if (!board) {
-      return;
-    }
-    setData(undefined);
-    setError(undefined);
-    const path = `${k8sBasePath}/api/v1/namespaces/openshift-monitoring/configmaps/grafana-dashboard-${board}`;
-    safeFetch(path)
+    safeFetch('/api/console/monitoring-dashboard-config')
       .then((response) => {
-        const json = _.get(response, ['data', `${board}.json`]);
-        if (!json) {
-          setError('Dashboard definition JSON not found');
-        } else {
-          const newData = JSON.parse(json);
-
-          _.each(newData?.templating?.list as TemplateVariable[], (v) => {
-            if (v.type === 'query' || v.type === 'interval') {
-              patchVariable(v.name, {
-                isHidden: v.hide !== 0,
-                options: _.map(v.options, 'value'),
-                query: v.type === 'query' ? v.query : undefined,
-                value: _.find(v.options, { selected: true })?.value || v.options?.[0]?.value,
-              });
-            }
-          });
+        setLoaded();
+        setError(undefined);
 
-          // Call this after creating the board's variables so we don't trigger a component update
-          // that will fail due to missing variables
-          setData(newData);
-        }
+        const getBoardData = (item) => ({
+          data: JSON.parse(_.values(item?.data)[0]),
+          name: item.metadata.name,
+        });
+        const newBoards = _.sortBy(_.map(response.items, getBoardData), (v) =>
+          _.toLower(v?.data?.title),
+        );
+        setBoards(newBoards);
       })
       .catch((err) => {
+        setLoaded();
         if (err.name !== 'AbortError') {
           setError(_.get(err, 'json.error', err.message));
         }
       });
-  }, [board, patchVariable, safeFetch]);
-
-  if (!board) {
-    return null;
-  }
-  if (error) {
-    return <ErrorAlert message={error} />;
-  }
-  if (!data) {
-    return <LoadingInline />;
-  }
+  }, [safeFetch, setLoaded]);
 
-  const rows = _.isEmpty(data.rows) ? [{ panels: data.panels }] : data.rows;
+  const boardItems = React.useMemo(() => _.mapValues(_.mapKeys(boards, 'name'), 'data.title'), [
+    boards,
+  ]);
 
-  return (
-    <>
-      {_.map(rows, (row, i) => (
-        <div className="row monitoring-dashboards__row" key={i}>
-          {_.map(row.panels, (panel, j) => (
-            <Card key={j} panel={panel} />
-          ))}
-        </div>
-      ))}
-    </>
-  );
-};
+  const changeBoard = (newBoard: string) => {
+    if (newBoard !== board) {
+      clearVariables();
 
-const MonitoringDashboardsPage_: React.FC<MonitoringDashboardsPageProps> = ({
-  clearVariables,
-  deleteAll,
-  match,
-  patchVariable,
-}) => {
-  const { board } = match.params;
+      const data = _.find(boards, { name: newBoard })?.data;
 
-  // Clear queries on unmount
-  React.useEffect(() => deleteAll, [deleteAll]);
+      _.each(data?.templating?.list as TemplateVariable[], (v) => {
```

Comment:
We should avoid `as` when possible. It usually means missing or wrong types. This is slightly better:

```suggestion
      _.each(data?.templating?.list, (v: TemplateVariable) => {
```

Do we have a type we can assign to `data`?

Reply:
Addressed by adding a `Board` type in PR https://github.com/openshift/console/pull/4454